en/guide/routing.md

```diff
@@ -68,19 +68,21 @@ app.all('/secret', (req, res, next) => {
 
 Route paths, in combination with a request method, define the endpoints at which requests can be made. Route paths can be strings, string patterns, or regular expressions.
 
-The characters `?`, `+`, `*`, and `()` are subsets of their regular expression counterparts. The hyphen (`-`) and the dot (`.`) are interpreted literally by string-based paths.
+{% capture caution-character %} In express 5, the characters `?`, `+`, `*`, `[]`, and `()` are handled differently than in version 4, please review the [migration guide](/{{ page.lang }}/guide/migrating-5.html#path-syntax) for more information.{% endcapture %}
 
-If you need to use the dollar character (`$`) in a path string, enclose it escaped within `([` and `])`. For example, the path string for requests at "`/data/$book`", would be "`/data/([\$])book`".
+{% include admonitions/caution.html content=caution-character %}
 
-<div class="doc-box doc-info" markdown="1">
+If you need to use the dollar character (`$`) in a path string, escape it with `\`. For example, the path string for requests at `/data/$book`, would be `/data/\$book`.
```

Comment:
You can just write a `$` in Express 5, there's no need to escape. The only characters needed to escape are values used by `path-to-regexp` specifically, and they're here: https://github.com/pillarjs/path-to-regexp/blob/d63f44bc54f2c6fe7d35dc77f8515622006bf7cb/src/index.ts#L105

Reply:
Thanks, I didn't know that this had changed. It might be good to inform about this change in the migration guide and in the README of path-to-regex

Reply:
It is under:

> Regexp characters are not supported

And this has been true since at least 1.x of `path-to-regexp` IIRC. I haven't had an issue with anyone misunderstanding how `path-to-regexp` works since then because it hasn't been a literal "path is a regex" for a long time, so it's easier to avoid confusing notes like not needing to escape certain characters. Easier to just stick with documenting the DSL that is used.

Reply:
Of course, that note could be clarified, since maybe it isn't clear that `^`, `$`, etc are regex characters?